### PR TITLE
Notations and convention

### PR DESCRIPTION
**Pull request by dericed**
Many `may, should, must, required, recommended` are changed to all
caps. If the usage of the work didn’t quite align with RFC2119’s
meanings then I reword to avoid using RFC2119’s terms. Please review
carefully.

### CODE REVIEW

File: codec_specs.md [hunk 1]
```diff
@@ -46,7 +46,7 @@ The private data contains the VCM structure BITMAPINFOHEADER including the extra
 
 | V_THEORA | Theora | The private data contains the first three Theora packets in order. The lengths of the packets precedes them. The actual layout is: 
 
-* Byte 1: number of distinct packets '`#p`' minus one inside the CodecPrivate block. This should be '2' for current Theora headers. 
+* Byte 1: number of distinct packets '`#p`' minus one inside the CodecPrivate block. This SHOULD be '2' for current Theora headers.
```
Comment (dericed) [inline codec_specs.md]: MUST?

**Comment by dericed** [comment]
This is branched from https://github.com/Matroska-Org/matroska-specification/pull/28

File: codec_specs.md (hunk 1 above)
Comment (mbunkus) [inline codec_specs.md]: As of 2016-07-08 it's a MUST, but it depends on the interpretation of "current", I guess. So yes, change it.

File: codec_specs.md [hunk 2]
```diff
@@ -133,7 +133,7 @@ For the samplerates 12kHz (00); 11,025kHz (01) and 8kHz (10) the BSID is 10
 | A_DTS/LOSSLESS | Digital Theatre System Lossless | DTS Lossless audio that does not have a core substream. The private data is void. Corresponding ACM wFormatTag : 0x2001 |
 
 | A_VORBIS | Vorbis | The private data contains the first three Vorbis packet in order. The lengths of the packets precedes them. The actual layout is:
-Byte 1: number of distinct packets '`#p`' minus one inside the CodecPrivate block. This should be '2' for current Vorbis headers.
+Byte 1: number of distinct packets '`#p`' minus one inside the CodecPrivate block. This SHOULD be '2' for current Vorbis headers.
```
Comment (mbunkus) [inline codec_specs.md]: Same case as Theora → "MUST"

File: diagram.md [hunk 3]
```diff
@@ -22,7 +22,7 @@ The first picture is a simple representation of a Matroska file.
...
 
-The [Metaseek]({{site.baseurl}}/index.html#MetaSeekInformation) section contains an index of where all of the other groups are in the file arelocated, such as the Track information, Chapters, Tags, Cues, Attachments, andso on. This element isn't technicaly required, but you would have to searchthe entire file to find all of the other Level 1 elements if you did not haveit. This is because any of the items can occur in any order. For instance youcould have the chapters section in the middle of the Clusters. This is partof the flexibility of EBML and Matroska.
+The [Metaseek]({{site.baseurl}}/index.html#MetaSeekInformation) section contains an index of where all of the other groups are in the file arelocated, such as the Track information, Chapters, Tags, Cues, Attachments, andso on. This element isn't technicaly REQUIRED, but you would have to searchthe entire file to find all of the other Level 1 elements if you did not haveit. This is because any of the items can occur in any order. For instance youcould have the chapters section in the middle of the Clusters. This is partof the flexibility of EBML and Matroska.
```
Comment (mbunkus) [inline diagram.md]: "Metaseek" → "MetaSeek"; Several words are missing spaces: "searchthe", "youcould", "partof", "haveit"

File: diagram.md [hunk 4]
```diff
@@ -180,19 +180,19 @@ While EBML allows elements of the same level to be in no particular order, for b
...
+The [Header]({{site.baseurl}}/index.html#EBMLHeader) MUST occur at the beginning of the file. This is how the library knows whether or not it can read the file. The design of EBML is pretty straight forward and is its own project on [SourceForge](http://ebml.sourceforge.net/). EBML is not just for Matroska, there are many different potential applications for it. As such, there is the possibility of there being new versions, such as a 2.0 design. The [EBMLVersion]({{site.baseurl}}/index.html#EBMLVersion) element would let the parser know first if it can read this file at all. If the EBMLVersion is set to 2.0, and the library is only able to read up to 1.2, then it knows it SHOULD NOT even attempt to read this file.
```
Comment (mbunkus) [inline diagram.md]: Change EBML project's URL to Github?

File: diagram.md [hunk 5]
```diff
@@ -180,19 +180,19 @@ While EBML allows elements of the same level to be in no particular order, for b
...
+The [MetaSeek]({{site.baseurl}}/index.html#MetaSeekInformation) section is to let the parser know where the other major parts of the file are. The design is pretty simple. Normally there is just one [SeekHead]({{site.baseurl}}/index.html#SeekHead) in a file. You then have a couple of [Seek]({{site.baseurl}}/index.html#Seek) entries. One for each seek point. The [SeekID]({{site.baseurl}}/index.html#SeekID) contains the "Class-ID" of a level 1 element. For example, the [Tracks]({{site.baseurl}}/index.html#Tracks) element has a Class-ID of "[16][54][AE][6B]". You would put that inthe SeekID, and then the byte position of that particular element in [SeekPosition]({{site.baseurl}}/index.html#SeekPosition). The Meta Seek section is usually just used when the file is openned so that it can get information about the file. Any seeking that happens when playing back the file uses the Cues.
```
Comment (mbunkus) [inline diagram.md]: Missing spaces: "inthe"

File: diagram.md (hunk 4 above)
Comment (dericed) [inline diagram.md]: Because of our deadline, I suggest to address this more comprehensively later. The URLs don't render in the text based RFC anyhow.

**Comment by dericed** [comment]
address all comments, and force pushed. Please re-review.

File: overhead.md [hunk 6]
```diff
@@ -135,7 +135,7 @@ We clearly see that there is another more interresting table (lacing use is drop
 *   In most cases, the best result is when 8 Blocks are packed in a Cluster. Bigger values don't improve the overhead much. It also seems to be a good value to stop using lacing in the low audio bitrate example. **So both lacing and clustering will be limited to 8 elements in libmatroska** on writing.
 *   In an agressive case like a 64kbps CBR video codec at 20 frames/sec can still achieve an overhead of less than 2% which is quite good.
 *   Upper 128kbps an overhead of less than 1% can always be achieved.
-*   Having 2 frames in a lace can improve substanctially the overhead. But as I frames should always have a proper timecode, it is not possible to use this solution.
+*   Having 2 frames in a lace can improve substanctially the overhead. But as I frames SHOULD always have a proper timecode, it is not possible to use this solution.
```
Comment (mbunkus) [inline overhead.md]: Typo: should be "substantially"

File: streaming.md [hunk 7]
```diff
@@ -10,18 +10,18 @@ In the case of Matroska, there are mostly 2 different kinds of stream: file acce
...
+File access can simply be reading a file located on your computer, but also accessing it from an HTTP (web) server or CIFS (windows share) server. All these protocol are usually safe from reading errors and seeking in the stream is possible. On other hand when the file is stored far or on a slow server, seeking can be an expensive operation and SHOULD be avoided. That's why we set a [few guidelines]({{site.baseurl}}/index.html) that, when followed, help reduce the number of seeking for regular playback and also have the playback start quickly without a lot of data needed to read first (like the Cues (index), Attachments or Meta Seek of all the Clusters).
```
Comment (mbunkus) [inline streaming.md]: "all these protocols" should be plural; "is stored far or…" → "is stored far away or…"

File: streaming.md [hunk 8]
```diff
@@ -10,18 +10,18 @@ In the case of Matroska, there are mostly 2 different kinds of stream: file acce
...
 
-A live Matroska stream is different than a file, because it may have no known end (only when the client disconnects). For that the Segment _must_ use the "unknow" size (all 1s in the size). The other option would be to concatenate Segments with known sizes one after the other. This solution allows a change of codec/resolution between each segment which can be useful in some cases (switch between 4:3 and 16:9 in some TV programs for example).
+A live Matroska stream is different than a file, because it may have no known end (only when the client disconnects). For that the Segment MUST use the "unknow" size (all 1s in the size). The other option would be to concatenate Segments with known sizes one after the other. This solution allows a change of codec/resolution between each segment which can be useful in some cases (switch between 4:3 and 16:9 in some TV programs for example).
```
Comment (mbunkus) [inline streaming.md]: `must use the "unknow" size` → missing an "n" in "unknown"

File: subtitles.md [hunk 9]
```diff
@@ -6,25 +6,25 @@ layout: default
...
+*   Start and stop timecodes that are used in a timecodes native storage format SHOULD be removed when being placed in Matroska as they could interfere if the file is edited afterwards. Instead, the Blocks timecode and Duration SHOULD be used to say when the timecode is displayed.
 *   Because a "subtitle" stream is actually just an overlay stream, anything with a transparency layer could be use, including video.
 
 # Images Subtitles
 
 
 The first image format that is a goal to import into Matroska is the VobSub subtitle format. This subtitle type is generated by exporting the subtitles from a DVD.
 
-The requirement for muxing VobSub into matroska is v7 subtitles (see first line of the .IDX file). If the version is smaller, you must remux them using the SubResync utility from VobSub 2.23 (or MPC) into v7 format. Generally it should not be a problem though, because any newly created subs will be in v7 format.
+The requirement for muxing VobSub into matroska is v7 subtitles (see first line of the .IDX file). If the version is smaller, you must remux them using the SubResync utility from VobSub 2.23 (or MPC) into v7 format. Generally any newly created subs will be in v7 format.
```
Comment (mbunkus) [inline subtitles.md]: Capitalize "Matroska", please.

File: subtitles.md [hunk 10]
```diff
@@ -132,7 +132,7 @@ SRT is perhaps the most basic of all subtitle formats.
 It consists of four parts, all in text..
 
 1\. A number indicating which subtitle it is in the sequence.
-2\. The time that the subtitle should appear on the screen, and then disappear.
+2\. The time that the subtitle appears on the screen, and then disappear.
```
Comment (mbunkus) [inline subtitles.md]: "disappear" is missing an s

File: tagging.md [hunk 11]
```diff
@@ -266,36 +266,36 @@ When a Tag is nested within another Tag, the nested Tag becomes an attribute
...
+Advanced-users application might let you put any tag in your file. But for the rest of the applications, they usually give you a basic list of tags you can use. Both have their needs. But it's usually a bad idea to use custom/exotic tags because <em>you</em> will probably be the only person to use this information even though everyone else could benefit from it. So hopefully when someone wants to put information in one's file, they will find an official one that fit them and hopefully use it ! If it's not in the list, this person can contact us any time for addition of such a missing tag. But it doesn't mean it will be accepted... Matroska files are not meant the become a whole database of people who made costumes for a film. A website would be better for that... It's hard to define what SHOULD be in and what doesn't make sense in a file. So we'll treat each request carefully.
```
Comment (mbunkus) [inline tagging.md]: Strip HTML tags: `<em>you</em>`

**Comment by mbunkus** [comment]
Alright, I'm done. You're right, this will take a lot of work to turn it into a proper spec and not an assorted set of vague web pages :)

**Comment by dericed** [comment]
I tried to keep up and just force pushed. Some of your suggestions need a more comprehensive application (stripping html and capitalize Matroska), but the specific notes are updated and pushed.

**Comment by mbunkus** [comment]
Yes, I figured. I just concentrated on getting rid of grammar and spelling mistakes at the end — and on whether or not the handling of may/must/required etc. seemed sensible.

**Comment by mbunkus** [comment]
Merged.

**Pull request merged**